packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -115,6 +121,54 @@ class DatePickerInput extends React.Component {
     }
   }
 
+  static getDerivedStateFromProps(props, state) {
+    // Formats the selected date into a consistent value and sets input state.
+    const format = DateUtil.getDateFormat(props.intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' }));
+    if (props.value !== state.prevDate) {
+      const tempDate = DateUtil.convertToISO8601(props.value, format);
+      return {
+        day: tempDate.substring(8, 10),
+        month: tempDate.substring(5, 7),
+        year: tempDate.substring(0, 4),
+        prevDate: props.value,
+      };
+    }
+    return null;
+  }
+
+  /**
+   * Moves focus to the correct input depending on date ordering.
+   * @param {string} inputValue - The value from the current input.
+   * @param {int} type - The input type, based on DateUtil.inputType.
+   * @param {string} dateVariant - Date variant based on the currently supported date
+   *                      orders: 'MM-DD-YYYY', 'DD-MM-YYYY', or 'YYYY-MM-DD'.
+   */
+  moveFocusOnChange(inputValue, type, dateVariant) {
+    if (dateVariant === 'MM-DD-YYYY') {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.MONTH) {
+          this.dayInput.focus();
+        } else {
+          this.yearInput.focus();
+        }
+      }
+    } else if (dateVariant === 'DD-MM-YYYY') {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.MONTH) {
+          this.yearInput.focus();
+        } else if (type === DateUtil.inputType.DAY) {
+          this.monthInput.focus();
+        }
+      }
+    } else if (dateVariant === 'YYYY-MM-DD') {
+      if (inputValue.length > 1 && type === DateUtil.inputType.MONTH) {
```

Comment:
Does this need to be `inputValue.length > 1` or can it be `inputValue.length === 2` to be consistent with the logics above?

Reply:
Changed! [0b496eb](https://github.com/cerner/terra-framework/pull/1004/commits/0b496eb729d1b0fc5851cc50d820b24a267a67ec)